aiotg/bot.py

```diff
@@ -78,18 +82,50 @@ def no_handle(mt):
             )
             self._process_updates(updates)
 
-    def run(self):
+    def webhook_loop(self, webhook_url, loop):
+        """
+        Starts aiohttp web server.
+        """
+        app = web.Application(loop=loop)
+
+        async def handle(request):
+            update = await request.json()
+            self._process_update(update)
+            return web.Response()
+
+        app.router.add_route('POST', urlparse(webhook_url).path, handle)
+        web.run_app(app, port=urlparse(webhook_url).port)
+
+    async def _set_webhook(self, webhook_url, **options):
+        """
+        Register you webhook url for Telegram service.
+
+        Additional documentation on https://core.telegram.org/bots/api#setwebhook
+        """
+        return await self.api_call(
+            'setWebhook',
+            url=webhook_url,
+            **options
+        )
+
+    def run(self, webhook_url=""):
         """
         Convenience method for running bots
 
         :Example:
 
         >>> if __name__ == '__main__':
-        >>>     bot.run()
+        >>>     bot.run(webhook_url="webhook_url='https://yourserver.com/webhooktoken")
+
+        Use empty string for webhook_url to switch to getUpdates mode
         """
         loop = asyncio.get_event_loop()
         try:
-            loop.run_until_complete(self.loop())
+            loop.run_until_complete(self._set_webhook(webhook_url))
```

Comment:
Second question is right.  May be I need to return explicit stop_webhook method with run_until_complete?